### DISCUSSION TITLE
Sending panes to specific windows/tabs

### DISCUSSION THREAD

**Discussion by CryoViking** [wezterm/wezterm#4833] [Q&A]
### What Operating System(s) are you running on?

macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

Yabai

### WezTerm version

20240118-061239-9181d010

### Ask your question!

Is it possible to send panes to specific tabs/windows?

I have a use case where in my layout, I have a 1:3 setup where I have my editor on the left most pane, and then the right half is split into 3 vertical panes.

I was wondering if it was possible to write, into my config, a keymap that could basically manage a tab list per pane, so that in my layout, I could cycle any given pane through an unique list for that "position" alone. Thus allowing a singular pane to be swapped out in a tab, leaving all the other panes alone.

I am aware that I can send panes to new tabs/windows but I haven't seen anything in the documentation for bringing them back to the original window/tab. Nor have I seen any discussion about this being a potential feature in the future. That said I could very easily just be blind.

**Comment by wez** [comment]
It's not accessible directly through lua at this time, but there are a couple of functions in the CLI:
* https://wezfurlong.org/wezterm/cli/cli/move-pane-to-new-tab.html
* https://wezfurlong.org/wezterm/cli/cli/split-pane.html (look for `--move-pane-id`)
